### DISCUSSION TITLE
How to get raw data from skiaImage?

### DISCUSSION THREAD

**Discussion by shi-yan** [rust-skia/rust-skia#564] [Q&A]
Hello,

I'm trying to create an UI system using skia to draw stuff on texture.

I saw the following from the example code:

```rust
        image.encode_to_data(EncodedImageFormat::PNG).unwrap()
```

But I don't want to encode to PNG, I want to access raw data. the original c++ code seems to have encode_to_data() without arguments to return the raw data. But the rust version doesn't have it. Should I use BMP for this purpose ?

I know skia can create textures directly, but I'm currently using a wgpu rust implementation (not dawn). So I have to manually create textures. 

Thanks

**Comment by pragmatrix** [comment]
I think `read_pixels` or `read_pixels_..with_context` is what you are looking for. You might need to pass the current `DirectContext` so that Skia is able to download the texture / image from the GPU.
